packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -169,11 +207,100 @@ export const useTreeViewItems: TreeViewPlugin<UseTreeViewItemsSignature> = ({
 
   const areItemUpdatesPrevented = React.useCallback(() => areItemUpdatesPreventedRef.current, []);
 
+  const addItems = ({
+    items,
+    parentId,
+    depth,
+    getChildrenCount,
+  }: AddItemsParameters<TreeViewBaseItem>) => {
+    if (items) {
+      const newState = updateItemsState({
```

Comment:
I feel like we should slit `updateItemsState` into 2:

1. A new method that handles the shared logic that you use in `updateItemsState` and in `addItems`
2. `updateItemsState` that is only used for the already existing use cases.

Because the method stats to be quite complex :grimacing:

Reply:
Not sure I get what you mean here 🤔 `updateItemsState`is used more or less the same with or without lazy loading. The one small difference is this bit that processes the `children` received in the `items` prop if lazy loading is enabled:
```ts
    if (!ignoreChildren) {
      item.children?.forEach((child) => processItem(child, depth + 1, id));
    }
```

Reply:
Maybe I'm over-thinking it :laughing: